### ISSUE TITLE
Error when notification via MQTT

### ISSUE THREAD

**Issue by Blobonat** [ScorpioBroker/ScorpioBroker#340]
Hi,

when creating a subscription with a MQTT endpoint, Scorpio throws an internal error during notification.

Tested with `scorpiobroker/all-in-one-runner:spring-kafka-latest`

Steps to reproduce:
1. Create Entity
```
curl --location --request POST 'http://localhost:9090/ngsi-ld/v1/entityOperations/create' \
--header 'Content-Type: application/json' \
--header 'NGSILD-Tenant: openiot' \
--header 'Link: <https://fiware.github.io/data-models/full-context.jsonld>; rel="http://www.w3.org/ns/json-ld#context"; type="application/ld+json"' \
--data-raw '[
    {
        "id": "urn:ngsi-ld:Vehicle:A100",
        "type": "Vehicle",
        "brandName": {
            "type": "Property",
            "value": "Mercedes"
        },
        "speed": {
            "type": "Property",
            "value": 80
        }
    }
]'
```
2. Create subscription with MQTT endpoint
```
curl --location --request POST 'http://localhost:9090/ngsi-ld/v1/subscriptions/' \
--header 'Content-Type: application/json' \
--header 'NGSILD-Tenant: openiot' \
--header 'Link: <https://fiware.github.io/data-models/full-context.jsonld>; rel="http://www.w3.org/ns/json-ld#context"; type="application/ld+json"' \
--data-raw '{
    "id": "urn:ngsi-ld:Subscription:mqtt",
    "type": "Subscription",
    "entities": [
        {
            "type": "Vehicle"
        }
    ],
    "watchedAttributes": [
        "speed"
    ],
    "throttling": 0,
    "notification": {
        "format": "normalized",
        "attributes": [
            "speed"
        ],
        "endpoint": {
            "uri": "mqtt://mosquitto:1883/test/topic",
            "accept": "application/json"
        }
    }
}'
```
3. Trigger Notification via attribute update
```
curl --location --request POST 'http://localhost:9090/ngsi-ld/v1/entityOperations/update' \
--header 'NGSILD-Tenant: openiot' \
--header 'Link: <https://fiware.github.io/data-models/full-context.jsonld>; rel="http://www.w3.org/ns/json-ld#context"; type="application/ld+json"' \
--header 'Content-Type: application/json' \
--data-raw '[
    {
        "id": "urn:ngsi-ld:Vehicle:A100",
        "type": "Vehicle",
        "speed": {
            "type": "Property",
            "value": 80,
            "observedAt": "2022-09-10T12:20:24Z"
        }
    }
]'
```
- Expected behavior: Notification via MQTT
- Actual behavior: No notification and following error of Scorpio
```
java.lang.IndexOutOfBoundsException: Index 0 out of bounds for length 0
	at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64) ~[na:na]
	at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70) ~[na:na]
	at java.base/jdk.internal.util.Preconditions.checkIndex(Preconditions.java:266) ~[na:na]
	at java.base/java.util.Objects.checkIndex(Objects.java:359) ~[na:na]
	at java.base/java.util.ArrayList.get(ArrayList.java:427) ~[na:na]
	at com.google.common.collect.AbstractMapBasedMultimap$WrappedList.get(AbstractMapBasedMultimap.java:773) ~[guava-29.0-jre.jar!/:na]
	at eu.neclab.ngsildbroker.commons.subscriptionbase.NotificationHandlerMQTT.sendReply(NotificationHandlerMQTT.java:59) ~[Commons-1.1.0-SNAPSHOT.jar!/:na]
	at eu.neclab.ngsildbroker.commons.subscriptionbase.BaseNotificationHandler.notify(BaseNotificationHandler.java:80) ~[Commons-1.1.0-SNAPSHOT.jar!/:na]
	at eu.neclab.ngsildbroker.commons.subscriptionbase.BaseSubscriptionService.sendNotification(BaseSubscriptionService.java:506) ~[Commons-1.1.0-SNAPSHOT.jar!/:na]
	at eu.neclab.ngsildbroker.commons.subscriptionbase.BaseSubscriptionService$2.run(BaseSubscriptionService.java:484) ~[Commons-1.1.0-SNAPSHOT.jar!/:na]
```

**Comment by amitrnec** [comment]
Hi,
Now we have fixed this issue #347 

Thanks

**Comment by ScorpioBroker** [comment]
merged will be in next release
